### ISSUE TITLE
(bug) AwsConfig ignored as of s3Retrieverv2 refactoring

### ISSUE THREAD

**Issue by david-alza** [thomaspoignant/go-feature-flag#1691]
### Observed behaviour

As of 1.24, passing in `AwsConfig` to s3Retrieverv2 won't actually get used. It gets overwritten at Init since the internal `downloader` attribute is nil. This results in our services not being able to initialize the feature flag client since we're setting up our own AwsConfig that's different from what `config.LoadDefaultConfig` will use. 

I believe the issue is the change of checking if `AwsConfig` [is nil previously](https://github.com/thomaspoignant/go-feature-flag/pull/1606/files#diff-72ad8ecde378ae7ac3bfb1dc7a83b19fdd0efaf5b7612b420dbc5c403611405fL66) to checking if [the downloader is nil now](https://github.com/thomaspoignant/go-feature-flag/pull/1606/files#diff-72ad8ecde378ae7ac3bfb1dc7a83b19fdd0efaf5b7612b420dbc5c403611405fR34). I'm not exactly certain though on the intention of the refactor though, so this might not be correct. 

### Expected Behavior

I'd expect passing in AwsConfig to result in that being used in s3retrieverv2.Retriever. Right now I don't see a path that this Config ever gets used. 

### Steps to reproduce

I don't have an independent reproduction at this time, but moving from 1.23 to 1.24 caused our calls to `ffclient.Init` to start failing with an error for: 

```impossible to retrieve the flags, please check your configuration: unable to download item from S3 "feature_flags.yaml", operation error S3: GetObject, resolve auth scheme: resolve endpoint: endpoint rule error, Invalid region: region was not a valid DNS name.```

This is because we set the region directly by calling `config.WithRegion("us-west-2")`, but that gets ignored since the retriever is only using `config.LoadDefaultConfig` and falling back to the ENV variable (I presume) which we don't set.

**Comment by thomaspoignant** [comment]
@david-alza thanks for raising this issue.

I will check in the next few days the issue and I will try to provide a fix.

**Comment by david-alza** [comment]
Thanks @thomaspoignant! The more I look at it, the more I think we're just missing this block that existed prior to the refactor: 

```
if s.AwsConfig == nil {
	cfg, err := config.LoadDefaultConfig(ctx)
	if err != nil {
		initErr = fmt.Errorf("impossible to init S3 retriever: %v", err)
		return
	}
	s.AwsConfig = &cfg
}
```

I can put up a PR in a second to take a stab at fixing this.

**Comment by thomaspoignant** [comment]
Yes, I just looked at it and I think you are right, I've removed that code by mistake.
Feel free to open a PR if you have a bit of time, if not I can take of this tomorrow.
